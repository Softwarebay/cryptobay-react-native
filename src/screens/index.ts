export { HomeScreen } from './HomeScreen/HomeScreen';
export { WalthoughtScreen } from './WalthoughtScreen/WalthoughtScreen';
export { Screens } from './screens.enum';
