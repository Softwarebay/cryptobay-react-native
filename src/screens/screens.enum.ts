export enum Screens {
  HOME = 'Home',
  WALTHOUGHT = 'Walthought',
  SWAP = 'Swap',
  SETTINGS = 'Settings',
}
