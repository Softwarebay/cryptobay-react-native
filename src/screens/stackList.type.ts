export type StackList = {
  Home: undefined;
  Settings: undefined;
  Swap: undefined;
  WalletSetup: undefined;
  Walthought: undefined;
};
