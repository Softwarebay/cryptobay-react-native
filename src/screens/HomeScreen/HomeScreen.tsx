import { NativeStackScreenProps } from '@react-navigation/native-stack';
import React from 'react';
import { SafeAreaView, Text } from 'react-native';

import { Screens } from '../screens.enum';
import { StackList } from '../stackList.type';

type HomeScreenProps = NativeStackScreenProps<StackList, Screens.HOME>;

export const HomeScreen: React.FC<HomeScreenProps> = (/* {
  navigaition,
}: HomeScreenProps */) => {
  return (
    <SafeAreaView>
      <Text>HomeScreen</Text>
    </SafeAreaView>
  );
};
