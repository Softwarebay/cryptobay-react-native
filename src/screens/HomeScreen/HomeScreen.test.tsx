import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Screens } from '../screens.enum';

import { HomeScreen } from './HomeScreen';

const mockNavigation = {
  navigate: jest.fn(),
};

describe('HomeScreen Screen', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <HomeScreen
          navigation={mockNavigation as any}
          route={Screens.HOME as any}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
