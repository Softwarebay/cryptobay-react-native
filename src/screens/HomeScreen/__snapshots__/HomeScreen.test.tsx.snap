// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Screen Component Snapshot 1`] = `
<RCTSafeAreaView>
  <Text>
    HomeScreen
  </Text>
</RCTSafeAreaView>
`;
